code/parachain/runtime/common/src/xcmp.rs

```diff
@@ -132,16 +132,19 @@ impl<
 		Err(XcmError::TooExpensive)
 	}
 
+	// how does this refunding works? - add docs
 	fn refund_weight(&mut self, weight: Weight) -> Option<MultiAsset> {
 		if let Some(ref asset_location) = self.asset_location {
 			let fee = WeightToFeeConverter::weight_to_fee(&weight);
 			let fee = self.fee.min(fee);
 			let price = fee.saturating_mul(self.price) / self.fee;
```

Comment:
multiply bar ration to be used here

Reply:
> // how does this refunding works? - add docs

See `drop` impl in the end of file. better group end to end fee/price/drop together if not.